### PR TITLE
Alma app creation

### PR DESCRIPTION
**Pull request by szmoon**
First pass at turning Trang's alma script into an application - I'm not 100% sure on what the endpoint requires and also don't have the ability to test it at the moment, so I kept the data being sent with the request the exact same:

```
{
  "holding_data": {
    "copy_id": "1"
  }, 
  "item_data": {
    "alternative_call_number": "", 
    "alternative_call_number_type": {
      "value": ""
    }, 
    "barcode": "", 
    "chronology_i": "", 
    "chronology_j": "", 
    "chronology_k": "", 
    "chronology_l": "", 
    "chronology_m": "", 
    "description": "Box 21", 
    "enumeration_a": "Box 21", 
    "enumeration_b": "", 
    "enumeration_c": "", 
    "enumeration_d": "", 
    "enumeration_e": "", 
    "enumeration_f": "", 
    "enumeration_g": "", 
    "enumeration_h": "", 
    "fulfillment_note": "", 
    "internal_note_1": "", 
    "internal_note_2": "", 
    "internal_note_3": "", 
    "is_magnetic": false, 
    "physical_material_type": {
      "value": "GRAPHIC"
    }, 
    "pieces": "1", 
    "policy": {
      "value": "PHOTOSTUDY"
    }
  }
}
```

### CODE REVIEW

File: src/csv_upload.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+				'internal_note_1','internal_note_2','internal_note_3'
+			]
+
+item_data_headings = ['barcode','physical_material_type','policy',
+				'enumeration_a','enumeration_b','enumeration_c','enumeration_d','enumeration_e',
+				'enumeration_f','enumeration_g','enumeration_h','chronology_i','chronology_j',
+				'chronology_k','chronology_l','chronology_m','description','alternative_call_number',
+				'alternative_call_number_type','pieces','fulfillment_note',
+				'internal_note_1','internal_note_2','internal_note_3'
+			]
```
Comment (abrin) [inline src/csv_upload.py]: Is one of these a subset of the other? If so can we share data?

File: src/csv_upload.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		'bibID': bibID,
+		"holdingID": holdingID,
+		'data': data
+	}
+
+
+def upload_file(bibID, holdingID, data):
+	# get env variables
+	ALMA_API_KEY = os.getenv("ALMA_API_KEY")
+	ALMA_BASE_URL = os.getenv("ALMA_BASE_URL","https://api-na.hosted.exlibrisgroup.com/almaws/v1/")
```
Comment (abrin) [inline src/csv_upload.py]: can we move these to the top, along with values that are more like constants

File: src/csv_upload.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+def process_file(csvFile):
+	with open(csvFile, "r") as file:
+		# get a csv reader object and skip first row (headings)
+		reader = csv.reader(file)
+		next(reader)
+		
+		# process each row
+		for row in reader:
+			print("A ROW!")
```
Comment (abrin) [inline src/csv_upload.py]: can we remove this?

File: src/csv_upload.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+			parsed_data = parse_row(row)
+			upload_file(**parsed_data)
```
Comment (abrin) [inline src/csv_upload.py]: generally, I'd avoid the ** expansion unless all of the values are really unknown, can we just used the named params?

File: csv/test_list.csv [hunk 5]
```diff
@@ -1,2 +1,3 @@
 ﻿bib_id,holding_id,copy_id,barcode,physical_material_type,policy,is_magnetic,enumeration_a,enumeration_b,enumeration_c,enumeration_d,enumeration_e,enumeration_f,enumeration_g,enumeration_h,chronology_i,chronology_j,chronology_k,chronology_l,chronology_m,description,alternative_call_number,alternative_call_number_type,pieces,fulfillment_note,internal_note_1,internal_note_2,internal_note_3
+9922450270001551,22115982840001551,1,,GRAPHIC,PHOTOSTUDY,,Box 21,,,,,,,,,,,,,Box 21,,,1,,,,
```
Comment (abrin) [inline csv/test_list.csv]: did you mean to add another test id?  If so, can e change some of the data?

**Comment by szmoon** [comment]
@abrin Refactored according to your comments, should be ready for review again!

**Pull request merged**